frontend/src/components/Header/SearchInput.js

```diff
@@ -1,96 +1,89 @@
 import { useSelector } from 'react-redux'
-import { Autocomplete, Box, InputAdornment, TextField, Typography } from '@mui/material'
+import { Autocomplete, InputAdornment, Stack, TextField, Typography } from '@mui/material'
 import { Close as CloseIcon, Search as SearchIcon } from '@mui/icons-material'
 import { searchListSelector } from '../../store/modules/search'
+import debounce from 'lodash/debounce'
 
 function SearchInput({ placeholder, onChange = () => null, onSelect = () => null, onHighlightChange }) {
   const [inputValue, setInputValue] = useState('')
   const [selected, setSelected] = useState(null)
-  const changeTimeoutRef = useRef(0)
+  const debouncedFuncRef = useRef(null)
   const options = useSelector(searchListSelector)
 
   const handleChange = e => {
-    clearTimeout(changeTimeoutRef.current)
     setInputValue(e?.target?.value)
-
-    changeTimeoutRef.current = setTimeout(() => {
-      onChange(e)
-    }, 500)
   }
 
-  const handleInputKeyDown = useCallback(
-    e => {
-      if ([38, 40].includes(e?.keyCode)) {
-        const optionNodesList = document.querySelectorAll('.MuiAutocomplete-popper [data-feature-id]')
-        const prevFocusedNode = document.querySelector('.MuiAutocomplete-popper .Mui-focused[data-feature-id]')
-        const prevFocusedIndex = +prevFocusedNode?.dataset?.optionIndex
-
-        let nextIndex = prevFocusedIndex + (e?.keyCode === 38 ? -1 : 1)
-        if (nextIndex === -1) {
-          nextIndex = optionNodesList.length - 1
-        } else if (nextIndex === optionNodesList.length) {
-          nextIndex = 0
-        }
-
-        const nextOptionNode = [...optionNodesList].find(node => +node?.dataset?.optionIndex === nextIndex)
-        const nextOption = options?.find(item => item?.id === nextOptionNode?.dataset?.featureId)
-
-        onHighlightChange(nextOption)
-      }
+  const handleHighlightChange = useCallback(
+    (event, option) => {
+      onHighlightChange(option)
     },
-    [options, onHighlightChange],
+    [onHighlightChange],
   )
 
-  const handleOptionMouseEnter = useCallback(
+  useEffect(() => {
+    onSelect(selected)
+  }, [onSelect, selected])
+
+  const handleKeyPress = useCallback(
     e => {
-      const featureId = e?.target?.dataset?.featureId
-      const option = options?.find(item => item?.id === featureId)
-      onHighlightChange(option)
+      if (e?.keyCode === 13) {
+        if (debouncedFuncRef.current?.cancel) {
+          debouncedFuncRef.current.cancel()
+        }
+
+        onChange(e?.target?.value)
+      }
     },
-    [options, onHighlightChange],
+    [onChange],
   )
 
   useEffect(() => {
-    onSelect(selected)
-  }, [onSelect, selected])
+    const debouncedFunc = debounce(() => {
+      onChange(inputValue)
+    }, 500)
+    debouncedFuncRef.current = debouncedFunc
+
+    debouncedFunc()
+    return () => {
+      debouncedFunc.cancel()
+    }
+  }, [inputValue, onChange])
 
   return (
     <Autocomplete
-      freeSolo
```

Comment:
I think we should keep `freeSolo` and remove `value` from from the `<Autocomplete>`. We're using it as a search field rather than a select element, so we don't need it to remember which option was selected. I would like the field to retain any text a user put in there, however.

Reply:
@justinanderson if we use freeSolo Mui doesn't show "no results" block. With current setup user can type anything and it's not being cleared nor replaced with selected value.